### PR TITLE
feat(command-palette): Provide hook for additional actions

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This allows hooks to provide additional command palette actions via a defined hook.

### CODE REVIEW

**Comment by priscilawebdev** [comment]
@EvanPurkhiser Could you maybe please write a description? not sure what you would like to achieve here 🧐

File: src/sentry/static/sentry/app/components/search/sources/commandSource.jsx [hunk 1]
```diff
@@ -117,9 +121,19 @@ class CommandSource extends React.Component {
...
+  <Hook name="command-palette:action">
+    {({hooks}) => <CommandSource {...props} actions={[...hooks, ...ACTIONS]} />}
+  </Hook>
+);
+
 const CommandSourceWithFeature = props => (
   <Access isSuperuser>
-    {({hasSuperuser}) => <CommandSource {...props} isSuperuser={hasSuperuser} />}
+    {({hasSuperuser}) => (
+      <React.Fragment>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/search/sources/commandSource.jsx:133]: Do you really need React.Fragment here?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/search/sources/commandSource.jsx:133]: 👍 not anymore, needed it for a split second when I was trying something else, but didn't remove. Thanks

**Comment by EvanPurkhiser** [comment]
This allows getsentry to provide actions in the command palette, which I wanted for https://github.com/getsentry/getsentry/pull/3714. There is no ticket, I just did this because it would be useful for the support team.

**Comment by priscilawebdev** [comment]
@EvanPurkhiser thanks for the description 👌🙂

File: src/sentry/static/sentry/app/components/search/sources/commandSource.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/search/sources/commandSource.jsx:133]: @EvanPurkhiser can we remove this and get this merged?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/search/sources/commandSource.jsx:133]: There was some other issue I was running into with this when the hook wasn't set. I'll investigate at some point, but would like to merge yeah

**Comment by EvanPurkhiser** [comment]
Cloising for now, will revisit in the future.

**Pull request closed without merging**